hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/record/AbstractNearCacheRecord.java

```diff
@@ -42,6 +42,7 @@
             AtomicLongFieldUpdater.newUpdater(AbstractNearCacheRecord.class, "recordState");
 
     protected long creationTime = TIME_NOT_SET;
+    protected int partitionId;
```

Comment:
I think `partitionId` can be accessed from different user threads so maybe should be `volatile`?